openapi/workday.payroll/client.bal

```diff
@@ -0,0 +1,326 @@
+    decimal timeout = 60;
+    # The choice of setting `forwarded`/`x-forwarded` header
+    string forwarded = "disable";
+    # Configurations associated with Redirection
+    http:FollowRedirects? followRedirects = ();
+    # Configurations associated with request pooling
+    http:PoolConfiguration? poolConfig = ();
+    # HTTP caching related configurations
+    http:CacheConfig cache = {};
+    # Specifies the way of handling compression (`accept-encoding`) header
+    http:Compression compression = http:COMPRESSION_AUTO;
+    # Configurations associated with the behaviour of the Circuit Breaker
+    http:CircuitBreakerConfig? circuitBreaker = ();
+    # Configurations associated with retrying
+    http:RetryConfig? retryConfig = ();
+    # Configurations associated with cookies
+    http:CookieConfig? cookieConfig = ();
+    # Configurations associated with inbound response size limits
+    http:ResponseLimitConfigs responseLimits = {};
+    # SSL/TLS-related options
+    http:ClientSecureSocket? secureSocket = ();
+|};
+
+# This is a generated connector for [WorkDay payroll REST API v2](https://community.workday.com/sites/default/files/file-hosting/restapi/index.html) OpenAPI specification.
+# The Payroll service enables you to access and manage payroll information,such as pay groups, payroll inputs, and tax rates.
+@display {label: "Workday Payroll", iconPath: "“resources/workday.payroll.svg”"}
+public isolated client class Client {
+    final http:Client clientEp;
+    # Gets invoked to initialize the `connector`.
+    # The connector initialization requires setting the API credentials.  Follow [this guide] https://credentials.workday.com/docs/authentication/ and obtain tokens.
+    #
+    # + clientConfig - The configurations to be used when initializing the `connector` 
+    # + serviceUrl - URL of the target service 
+    # + return - An error if connector initialization failed 
+    public isolated function init(ClientConfig clientConfig, string serviceUrl) returns error? {
+        http:Client httpEp = check new (serviceUrl, clientConfig);
+        self.clientEp = httpEp;
+    }
+    # Retrieves a single pay group instance.
+    #
+    # + iD - The Workday ID of the resource. 
+    # + subresourceID - The Workday ID of the subresource. 
+    # + return - Successful response. A successful response can return no matched data. 
+    remote isolated function getPayGroupBySubResourceID(string iD, string subresourceID) returns PayGroupViewDetail|error {
+        string path = string `/jobs/${iD}/payGroup/${subresourceID}`;
+        PayGroupViewDetail response = check self.clientEp->get(path, targetType = PayGroupViewDetail);
+        return response;
+    }
+    # Retrieves a single pay group instance.
+    #
+    # + iD - The Workday ID of the resource. 
+    # + return - Successful response. A successful response can return no matched data. 
+    remote isolated function getPayGroupByID(string iD) returns PayGroupViewDetail|error {
```

Comment:
`string iD` is odd to use as a parameter name. Shall we rename the parameter name to `id` as below in OpenAPI file ?
```
      - name: id
        in: path
        description: The Workday ID of the resource.
        required: true
        schema:
          type: string
```